### ISSUE TITLE
Add possibility to start server when usb cable is plugged in

### ISSUE THREAD

**Issue by godfuture** [bk138/droidVNC-NG#40]
Hi,
my phone display was broken due to water. After some time, the display came back, but I guess this was simply luck. In case you want to have a backup when display is damaged, it would help to start VNC server only when USB is connected.

Still, if someone wants the server to be started all the time, the option requested here could be disabled.

Many thanks

**Comment by gitgrub** [comment]
If you just want to copy files from your phone, use adb pull/push commands. Would that work for this case?

**Comment by godfuture** [comment]
Not really, main focus is to operate the phone when the display is not available. VNC is a great assist in such cases. For example to save the phone in urgency case like activating USB debugging (very important) or to do backups or using authentication apps (exception: Microsoft Authenticator showed only black screen. Do you know why?). Or simply to install other apps to help with whatever problems occur.

If the server is always started, this is not only a security problem, but also costs battery. On usb cable, energy does not play a very big role.

**Comment by godfuture** [comment]
> Concerning the usb plug/unplug attempts, a way to go could be to start the server via adb command when the usb connection is stable.

Nice idea. I guess there is no usb debugging required?!

**Comment by bk138** [comment]
Would https://stackoverflow.com/questions/13380590/is-it-possible-to-start-activity-through-adb-shell be it? I don't know though if usb debugging would need to be enabled or not...

**Comment by gitgrub** [comment]
I think usb debugging has to be enabled to use adb. Look at https://developer.android.com/studio/command-line/adb
Here is an example list of adb commands, not complete for sure, but to see a bit whats possible:
https://gist.github.com/Pulimet/5013acf2cd5b28e55036c82c91bd56d8

**Comment by Martii** [comment]
> I think usb debugging has to be enabled to use adb.

It does on my phone... Galaxy S8. Newer ones have USB debugging over wifi option *(from articles I read when I first discovered this project)* but this one you have to connect to USB first then can use over wifi after a `adb tcp port`.

However if Android lets you detect/read the ARP, a list of "acceptable" MAC addresses to start/stop in droidVNC-NG could be used to start and stop the server just using TCP/IP... I haven't twiddle with that much but it might be an idea for better security.

**Comment by bk138** [comment]
Can now be done via automation apps like [MacroDroid](https://www.macrodroid.com/), [Automate](https://llamalab.com/automate/) or [Tasker](https://tasker.joaoapps.com/), see https://github.com/bk138/droidVNC-NG#remote-control-via-the-intent-interface
